src/components/feedback/FeedbackDirective.js

```diff
@@ -19,178 +22,176 @@ goog.require('ga_permalink');
-                  method: method,
-                  url: feedbackUrl,
-                  data: formData
-                };
-                var resetF = function() {
-                  scope.error = false;
-                  scope.success = false;
-                  scope.form = false;
-                  scope.showProgress = false;
-                  scope.file = null;
-                  scope.feedback = '';
-                };
-
-
-                if (!scope.isIE || gaBrowserSniffer.msie > 9) {
-                  params.transformRequest = angular.identity;
-                  params.headers = {'Content-Type': undefined};
-                } else {
-                  params.headers = {'Content-Type':
-                      'application/x-www-form-urlencoded'};
-                }
-
-                scope.showProgress = true;
-                $http(params).then(function() {
-                  resetF();
-                  scope.success = true;
-                }, function() {
-                  resetF();
-                  scope.error = true;
-                });
-
-              };
-            }
-          };
+        scope.permalinkValue = gaPermalink.getHref();
+
+        // Listen to permalink change events from the scope.
+        scope.$on('gaPermalinkChange', function(event) {
+          scope.permalinkValue = gaPermalink.getHref();
+        });
+
+        scope.$on('gaDrawingLayer', function(event, data) {
+          drawingLayer = data;
         });
+
+        scope.$watch('active', function(active) {
+          scope.form = true;
+          scope.error = false;
+          scope.success = false;
+
+          // Active draw only on big screen
+          scope.showDraw = active && gaWindow.isWidth('>s') &&
+              gaWindow.isHeight('>s');
+          if (drawingLayer) {
+            drawingLayer.getSource().clear();
+          }
+        });
+
+        scope.submit = function() {
+          var contentType;
+          var formData = createFormData();
+          var params = {
+            method: method,
+            url: feedbackUrl,
+            data: formData
+          };
+          var resetF = function() {
+            scope.error = false;
+            scope.success = false;
+            scope.form = false;
+            scope.showProgress = false;
+            scope.file = null;
+            scope.feedback = '';
+          };
+
+
+          if (!scope.isIE || gaBrowserSniffer.msie > 9) {
+            params.transformRequest = angular.identity;
+            params.headers = {'Content-Type': undefined};
+          } else {
+            params.headers = {'Content-Type':
+                'application/x-www-form-urlencoded'};
+          }
+
+          scope.showProgress = true;
+          $http(params).then(function() {
+            resetF();
+            scope.success = true;
+          }, function() {
+            resetF();
+            scope.error = true;
+          });
+        };
+      }
+    };
+  });
```

Comment:
Not related to this PR, but detected a small bug from the feedback dialog (it's like this in prod as well).

When I draw some elements from the feedback dialog, then I select one and delete it (with delete key), the object is not deleted directly. It's deleted only when clicking somewhere in the map.

Deletion is immediate in the drawing dialog (after confirmation), and it should behave the same from the feedback dialog.

Could you add this to this PR?